### ISSUE TITLE
False-positive cache miss when value is falsey

### ISSUE THREAD

**Issue by angeloashmore** [asyarb/use-static-query#1]
`useStaticQuery` uses a falsey check to determine if the cache has a value for a given key. If the value is falsey (e.g. `false`, `0`, `''`), the hook will determine the cache does not have a value for the key, even though it does.

https://github.com/asyarb/use-static-query/blob/16cdc307636be032ba353748eb2e7da99e3ba870/src/hook.tsx#L16-L17

Rather than using `!cache.get(cacheKey)` to determine if a cache value exists for a key, using something like `cache.hasOwnProperty(cacheKey)` or using a `Map` with `Map.prototype.has` will produce more sound results.

TL;DR `false`, `0`, `''`, etc. are valid cache values, but the hook does not think so.

**Comment by asyarb** [comment]
This was resolved in #3 being merged.
